pkg/scalers/nsq_scaler.go

```diff
@@ -0,0 +1,358 @@
+package scalers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+
+	"github.com/go-logr/logr"
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+)
+
+type nsqScaler struct {
+	metricType v2.MetricTargetType
+	metadata   nsqMetadata
+	httpClient *http.Client
+	logger     logr.Logger
+}
+
+type nsqMetadata struct {
+	NSQLookupdHTTPAddresses  []string `keda:"name=nsqLookupdHTTPAddresses,  order=triggerMetadata;resolvedEnv"`
+	Topic                    string   `keda:"name=topic,                    order=triggerMetadata;resolvedEnv"`
+	Channel                  string   `keda:"name=channel,                  order=triggerMetadata;resolvedEnv"`
+	DepthThreshold           int64    `keda:"name=depthThreshold,           order=triggerMetadata;resolvedEnv, default=10"`
+	ActivationDepthThreshold int64    `keda:"name=activationDepthThreshold, order=triggerMetadata;resolvedEnv, default=0"`
+
+	triggerIndex int
+}
+
+const (
+	nsqMetricType = "External"
+)
+
+func NewNSQScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %w", err)
+	}
+
+	logger := InitializeLogger(config, "nsq_scaler")
+
+	nsqMetadata, err := parseNSQMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing NSQ metadata: %w", err)
+	}
+
+	return &nsqScaler{
+		metricType: metricType,
+		metadata:   nsqMetadata,
+		httpClient: kedautil.CreateHTTPClient(config.GlobalHTTPTimeout, true),
```

Comment:
The `true` here is to ignore the TLS certificate verification. We should expose the field to users and set it `false` as default

Reply:
I added the `unsafeSsl` field to help enable this. I was also just using `http` previously, so it seemed like I needed to add something like `useHttps` as well.